### DISCUSSION TITLE
New integration: Baby Buddy for Android app release

### DISCUSSION THREAD

**Discussion by MrApplejuice** [babybuddy/babybuddy#626] [Show and tell]
Hello again cdubz!

Just an heads-up and maybe something for ["the list"](https://github.com/babybuddy/babybuddy#baby-buddy-on-the-web): 

I finally was able to publish the first version of the Android app for BabyBuddy that I was working on for a year now which allows for quicker tracking of timers in BabyBuddy. It is available for free from GitHub via the Releases-page - or for 2 bucks on the Android app store. Links:

- GitHub repository: https://github.com/MrApplejuice/BabyBuddyAndroid/
- Play store link: https://play.google.com/store/apps/details?id=eu.pkgsoftware.babybuddywidgets

Development time was quite slow due to - well -  caring for a newborn over that time interval and me not knowing how many hoops there are to get something published on the Android App Store (not too bad but also not "free"). The app is a bit behind on features, need to work on the new timer-types and tagging :laughing: 

Thanks a lot for the awesome website by the way! We had to track a few things for our newborn in the beginning for some medical reasons and the website was a great help for that! I plan to develop the app a bit further in the near future and continue to contribute to the app's ecosystem/success that way...

And I wanted to at least notify you about the app release myself!

Kind regards,
  Paul K.

------------------

Something else I cannot quite find back: I think you used to have a link to some hosting provider where you could host baby buddy directly. Was that removed? I am not sure if you received a kickback from that or so, but I would be happy to place a link to the hosting provider there if that still exists and makes sense? For the app it would make sense to at least direct someone to an easy-to-find/easy-to-use place to host baby buddy at. Maybe you know a place where people should go for that...?

**Comment by cdubz** [comment]
Exciting!! Congrats!

I haven't tested the play store link yet but I with the direct APK download I am getting an error message, "App not installed as package appears to be invalid".

Will definitely add this to BBoTW list.

> Something else I cannot quite find back: I think you used to have a link to some hosting provider where you could host baby buddy directly. Was that removed? 

Yes it was 😦 It was a Heroku click to install button but I abandoned it after Heroku pulled their support for free tier developer plans. Wasn't getting a kickback or anything and did not want to support that move.

> For the app it would make sense to at least direct someone to an easy-to-find/easy-to-use place to host baby buddy at. Maybe you know a place where people should go for that...?

Yes... that would be nice. And to be fair Heroku _is_ still easy to use. Much easier than other alternatives I looked at a few months back. I think it would be good to dedicate a page in the documentation to easy deployment scenarios like Heroku. Lets consider that work to be done based on this issue.

**Comment by cdubz** [comment]
Oh also meant to mention -- the original logo assets are available under the same license as Baby Buddy itself. Feel free to use them in your Android app if ya want.

https://github.com/babybuddy/babybuddy/tree/master/babybuddy/static_src/logo_assets

**Comment by MrApplejuice** [comment]
:blush: I was about to go to bed, but I could not leave a "broken apk" on the releases page after posting this. Should be fixed now! You can try again, I updated the APK: https://github.com/MrApplejuice/BabyBuddyAndroid/releases/tag/v1.1.5 The signing key was wrong, now you should only get a big old android warning that this is an untrusted app and then selected _Install Anyway_. Hope it works!

> the original logo assets are available under the same license as Baby Buddy itself

Oh wow, thanks a lot. I did not want to copy it verbatim. I think you can see that I at least got some "inspiration" from the original logo. Also the webapp-logo has some problems on Mobile where the icon is downscaled to smaller resolutions and looses too much detail at places (48x48 is the minimum resolution that needs to be supported for the launcher). That is why I simplified the icon a bit. But if you would like it, I can adapt the icon again and make it more similar to the webapp logo, for example set the right colors? Not sure if you would like that?

I felt kind of awkward just copying the "Baby Buddy" name itself and tagging on "for Android". But I came to understand that there are a lot of "baby buddies" out there already and that leaving the naming somewhat similar makes more sense for communication. Again, if you have any objections regarding the naming, please tell me.

**Comment by cdubz** [comment]
> But if you would like it, I can adapt the icon again and make it more similar to the webapp logo, for example set the right colors? Not sure if you would like that?

Entirely up to you! Just wanted to make sure you know you can use the assets 😄 

> I felt kind of awkward just copying the "Baby Buddy" name itself and tagging on "for Android". But I came to understand that there are a lot of "baby buddies" out there already and that leaving the naming somewhat similar makes more sense for communication. Again, if you have any objections regarding the naming, please tell me.

No objections at all!

Although this does make me wonder if I _should_ formalize the app name in some way 🤔 If only because, what if someone else did? Or someone used it for something legitimately malicious?

It's a weird conundrum as I want the whole ecosystem to be as flexible and welcoming as possible. I love to see the cool stuff people build around Baby Buddy!

**Comment by cdubz** [comment]
And confirming the updated APK works 👍  I'll poke around with this soon. Congrats, again!

**Comment by cdubz** [comment]
@MrApplejuice also curious -- are you interested in transferring the Android app to the Baby Buddy GH org and working on it there? No worries if not, understand if you want to keep it under your own account 😄

**Comment by MrApplejuice** [comment]
Hey there! I had to deal with some documentation issues reported by Google Play, so I was a bit caught up with that...

> Although this does make me wonder if I should formalize the app name in some way thinking If only because, what if someone else did? Or someone used it for something legitimately malicious?

I am certainly not an expert on legal issues, especially international legal issues. But yes, I think that might be a good idea for a fairly popular project like this (1.6k stars!!). Maybe something like "media use allowed under [insert license here]" and under the following circumstances [...] could be enough already? Not sure, but even Linux is [registered as a trademark](https://www.linuxfoundation.org/legal/trademark-usage) to avoid issues. I would guess to prevent name poaching or sth... but I am really no expert at this either, sorry.

> And confirming the updated APK works +1 I'll poke around with this soon. Congrats, again!

Thanks! Also again for immediately testing the APK and finding the issue! I am very busy with multiple improvements wrt what events can be tracked and tagging-support, but right now development really has slowed down, because Google Play requires some baby sitting - and of course there were some issues that were revealed after an open release like this...

> are you interested in transferring the Android app to the Baby Buddy GH org and working on it there? No worries if not, understand if you want to keep it under your own account smile

If you are interested to have this app "in the fold", we can do that! Because of my legal name-connection with the app on the Android-store, I need to insist on full control over the repository, however. If that is okay with you, we can migrate the repository, for sure! I think that could be a great idea for the project and increase its visibility.

But maybe you also want to double check if the app actually goes in a direction you like? I am trying to develop some updates, but cannot promise a timeline or anything. I am not sure if there are expectations from your side for what should happen if a project is added to your organization? (quality standards, update, community, etc.)

**Comment by cdubz** [comment]
> I am certainly not an expert on legal issues, especially international legal issues. But yes, I think that might be a good idea for a fairly popular project like this (1.6k stars!!). Maybe something like "media use allowed under [insert license here]" and under the following circumstances [...] could be enough already? Not sure, but even Linux is [registered as a trademark](https://www.linuxfoundation.org/legal/trademark-usage) to avoid issues. I would guess to prevent name poaching or sth... but I am really no expert at this either, sorry.

Yeah, to be clear those were just thinking out loud questions... I wasn't expecting answers 😄 I did do a little research on it and I'm still a bit meh on the topic. I think it makes more sense with a backing _group_ as opposed to individual maintainer. Explicitly stating how this applies to media somewhere may be a good enough change.

> If you are interested to have this app "in the fold", we can do that! Because of my legal name-connection with the app on the Android-store, I need to insist on full control over the repository, however. If that is okay with you, we can migrate the repository, for sure! I think that could be a great idea for the project and increase its visibility.

Great! Lets do it... I think you initiate that on your end? I can't remember what the transfer process looks like exactly. Let me know if you need any guidance.

> But maybe you also want to double check if the app actually goes in a direction you like? I am trying to develop some updates, but cannot promise a timeline or anything. I am not sure if there are expectations from your side for what should happen if a project is added to your organization? (quality standards, update, community, etc.)

This free time, open source development -- I have no expectations! I just think it's great you're working on this now. Whether you are able to in the future... what can ya do? 🤷

**Comment by MrApplejuice** [comment]
> Whether you are able to in the future... what can ya do?

Okay, then we are on the same page here :smile: 

I joined the organization, thanks for the invite! Please give me a a week or so for the transition. I am considering if I want to recreate the repository from scratch. I know - sounds like a "crazy" idea for doing the migration, but I made some mistakes at the very start that I _could_ maybe reconcile now (had to use `git filter-branch` to correct a mistake but that messed up other things). Need to think about a strategy here ... 

> Great! Lets do it... I think you initiate that on your end? [...migration...]

Yes, I know how it works. Had to do it many times for my 9-5. Maybe permissions will be relevant down the line, but I will tell you if I hit a wall!

**Comment by MrApplejuice** [comment]
Okay... I got a plan for the migration: I want to focus on fixing some bugs with the apps and pushing a major improvement of the app first right now. After I released the set of updates, I will migrate the repository to the babybuddy-org! Will still take a few weeks but I suppose that in May, it is time to get packing and move over :wink:

**Reply by MrApplejuice** [reply]
Wow... last update was... April 9, literally half a year ago. Anyway... the big feature milestone from my side is almost complete:

https://github.com/MrApplejuice/BabyBuddyAndroid/milestone/4

Last issue standing is to do the actual move of the repository. I am getting things ready right now to move over, like:

- Collecting strings in the application used for back-referencing to the GitHub presence
- Getting a grip on the Android store presence to fix all the back references there as well

This probably takes a bit, still, but I am planning to do a new feature-release of the app within the next week. This will then include the migration to https://github.com/babybuddy/BabyBuddyAndroid (naming discussion, see below). The move will be done with the GitHub-transfer ownership feature!

--------------------------

I would like to ping you, @cdubz, over the move one last time to check some things I would rather get right the first time around though:

- I thought about the naming of the repository a bit, for example. `BabyBuddyAndroid` clashes a bit with the pattern you have going on in the organization: https://github.com/babybuddy . `babybuddy-for-android` would probably be a better name? Renaming the repo might be a bit of a rougher ride for the migration because it requires two "renaming steps" here on GitHub. I do not know if GitHub handles redirect in that case correctly for `MrApplejuice/BabyBuddyAndroid -> babybuddy/BabyBuddyAndroid -> babybuddy/babybuddy-for-android`. Any idea/preference?
- Other than that, something else you would want to provide input on?

**Comment by MrApplejuice** [comment]
@cbudz I moved the repo to the babybuddy organization, got a quick question though: could you make me an admin for the project https://github.com/babybuddy/babybuddy-for-android again? I would like to be able to configure some of the options... :-)

**Reply by cdubz** [reply]
Yes, of course. Done!

**Comment by MrApplejuice** [comment]
@cbudz

I wanted to audit my github-release strategy for the android-app today and have a quick look if I am not exhausting the available data-transfer limits, or storage capacity for that matter. But of-course, the app now is part of the babybuddy-org, where I do not have access to those metrics:

![image](https://github.com/babybuddy/babybuddy/assets/542105/f2702f23-4dae-4f01-883f-a081f426d573)
(this is from my "personal account")

I am not sure what the quota are. Background: I am considering integrating some GitHub Actions-based CI/CD, but I have no clue right now what potential limits or constraints are that I operate under. Can you reveal some of those metrics maybe? I do not want to mess up babybuddy's CI/CD as part of the org. So having _some_ metrics would be very handy!

**Reply by cdubz** [reply]
@MrApplejuice I made you an owner of the org so you can poke around. GitHub recommends at least two owners for an org anyway (:

Let me know if you think anything is insufficient. I *think* we can apply for some sort of higher tier as a OSS project? I'm not sure.

**Reply by MrApplejuice** [reply]
Oh wow wow... ehm... thank you for your trust in me, a random person on the internet! If you would like to have something from my side, I am happy to share some credentials with you for the sake of trust-building. In the vein of a professor I once had who said that he only trusts someone to manage his servers if he knows where to go to "throw a stone through the person's window".

-----

The for something I do not quite understand... I see zero GitHub actions-minutes on the babybuddy-org. See here: https://github.com/organizations/babybuddy/settings/billing/summary but there were definitely GitHub actions being used over the past week. Can you see something on your end? Sorry to ask for this one more time, but I somehow assume that I cannot see the _current_ stats as a freshly-baked owner, maybe?

> I think we can apply for some sort of higher tier as a OSS project?

If the stats that I see are correct, the babybuddy-org is using nothing(??) which seems to give plenty of headway for some CI/CD of the android app. But I still do not think that I have up-to-date stats here. Also it could indicate that the org already is treated "specially" by GitHub. You created a pretty popular app afterall!

![image](https://github.com/babybuddy/babybuddy/assets/542105/9d82c871-ffda-403a-882b-b9f226acb538)

**Reply by cdubz** [reply]
That is what I see as well 🤷 In fact when I recently started working in GitHub on some _private_ repos I didn't even realize these things had limits haha.

Lets move forward assuming we have space to grow and we can keep an eye on things.

**Reply by MrApplejuice** [reply]
> Lets move forward assuming we have space to grow and we can keep an eye on things.

Sounds good to me! I was trying to read up on limits in the docs. I know for a fact that the limits apply to private repos from my work. But maybe that is it then: public repos have, at "generous" limits that do not count towards the billed private-repo minutes? From what I see this is the most likely solution to the puzzle.

Time to put this theory to the test by provisioning some super-heavy Java build toolchain for cross-compiling code to Android... :scream:
